runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/test/aws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpResponseTest.kt

```diff
@@ -71,114 +67,4 @@ class OkHttpResponseTest {
 
         assertContentEquals(content, actualBody)
     }
-
-    @Test
-    fun testCallContextCancelsWriter() = runTest {
-        // call context cancelled should result in no children left
-        val request = HttpRequest(HttpMethod.GET, Url.parse("https://aws.amazon.com"), Headers.Empty, HttpBody.Empty)
-
-        val execContext = ExecutionContext()
-        val callJob = Job(coroutineContext.job)
-        val callContext = coroutineContext + callJob
-        val okRequest = request.toOkHttpRequest(execContext, callContext)
-
-        // something sufficiently large enough that a single write doesn't cause the coroutine to immediately complete
-        val content = ByteArray(16 * 1024 * 1024)
-
-        val okResponse = Response.Builder().apply {
-            protocol(Protocol.HTTP_1_1)
-            code(200)
-            body(content.toResponseBody())
-            message("OK")
-            addHeader("foo", "bar")
-            request(okRequest)
-        }.build()
-
-        val sdkResponse = okResponse.toSdkResponse(callContext)
-        yield()
-
-        val body = sdkResponse.body
-        assertIs<HttpBody.Streaming>(body)
-        val ch = body.readFrom()
-        assertFalse(ch.isClosedForWrite)
-
-        assertEquals(1, callContext.job.children.toList().size)
-        callJob.cancel()
-        callJob.complete()
-        callJob.join()
-        assertEquals(0, callContext.job.children.toList().size)
-        assertTrue(ch.isClosedForWrite)
-    }
-
-    @Test
-    fun testSourceReadError(): Unit = runBlocking {
```

Comment:
**Question**: Why is this test no longer necessary?

Reply:
This test was exercising that any exception in the background coroutine resulted in the exception being propagated through the channel (and then later also testing that the exception didn't bubble up which you added recently). 

Neither is necessary now as callers will be reading directly from the underlying OkHttp body. Exceptions will be thrown transparently and there is no cancellation/propagation involved anymore.